packages/common/src/search/_internal/hubEventsHelpers/processFilters.ts

```diff
@@ -5,21 +5,42 @@ import {
 } from "../../../events/api/orval/api/orval-events";
 import { getOptionalPredicateStringsByKey } from "./getOptionalPredicateStringsByKey";
 import { getPredicateValuesByKey } from "./getPredicateValuesByKey";
+import { IDateRange } from "../../types/types";
+import { searchGroups } from "@esri/arcgis-rest-portal";
+import { IHubRequestOptions } from "../../../types";
+import { isUpdateGroup } from "../../../utils/is-update-group";
 
 /**
  * Builds a Partial<GetEventsParams> given an Array of IFilter objects
  * @param filters An Array of IFilter
  * @returns a Partial<GetEventsParams> for the given Array of IFilter objects
  */
-export function processFilters(filters: IFilter[]): Partial<GetEventsParams> {
+export async function processFilters(
+  filters: IFilter[],
+  requestOptions: IHubRequestOptions
+): Promise<Partial<GetEventsParams>> {
   const processedFilters: Partial<GetEventsParams> = {};
   const access = getOptionalPredicateStringsByKey(filters, "access");
   if (access?.length) {
-    // TODO: remove ts-ignore once GetEventsParams supports filtering by access
-    // @ts-ignore
     processedFilters.access = access;
   }
-  const term = getPredicateValuesByKey(filters, "term");
+  const canEdit = getPredicateValuesByKey<boolean>(filters, "canEdit");
+  if (canEdit.length) {
+    processedFilters.canEdit = canEdit[0].toString();
+  }
+  const entityIds = getOptionalPredicateStringsByKey(filters, "entityId");
+  if (entityIds?.length) {
+    processedFilters.entityIds = entityIds;
+  }
+  const entityTypes = getOptionalPredicateStringsByKey(filters, "entityType");
+  if (entityTypes?.length) {
+    processedFilters.entityTypes = entityTypes;
+  }
+  const eventIds = getOptionalPredicateStringsByKey(filters, "eventId");
```

Comment:
Just realized I had only committed the documentation changing this from `eventId` to `id`. Lets tweak this to:

```ts
const eventIds = getOptionalPredicateStringsByKey(filters, "id");
if (eventIds?.length) {
  processedFilters.eventIds = eventIds;
}
```